packages/gen-ai-hub/src/client/openai/openai-client.ts

```diff
@@ -21,22 +25,17 @@ const apiVersion = '2024-02-01';
 export class OpenAiClient {
   /**
    * Creates a completion for the chat messages.
-   * @param model - The model to use for the chat completion.
    * @param data - The input parameters for the chat completion.
-   * @param deploymentResolver - A deployment id or a function to retrieve it.
+   * @param deploymentConfig - This configuration is used to retrieve a deployment. Depending on the configuration use either the given deployment ID or the model name to retrieve matching deployments. If model and deployment ID are given, the model is verified against the deployment.
    * @param requestConfig - The request configuration.
    * @returns The completion result.
    */
   async chatCompletion(
-    model: OpenAiChatModel | { name: OpenAiChatModel; version: string },
     data: OpenAiChatCompletionParameters,
-    deploymentResolver?: DeploymentResolver,
+    deploymentConfig: DeploymentConfiguration<OpenAiChatModel>,
```

Comment:
[pp] Declaring this `DeploymentConfiguration` for me puts the focus on the deployment, which is rather a detail we would want to hide from users for most use cases. Instead, I'd prefer the API to naturally hint towards the models.

Reply:
Makes sense. I don't have the greatest ideas for renaming though. `ModelConfiguration` is not exact enough as it excludes the deployment ID. Just putting some ideas out there: `ModelDeployment`, `DeployedModelConfig`, something with an "or" inbetween, but I don't like that.

Reply:
@deekshas8 @tomfrenken Any suggestions?